dcm4che-imageio/src/main/java/org/dcm4che3/imageio/plugins/dcm/DicomAccessor.java

```diff
@@ -0,0 +1,98 @@
+//
+///////////////////////////////////////////////////////////////
+//                C O P Y R I G H T  (c) 2019                //
+//        Agfa HealthCare N.V. and/or its affiliates         //
+//                    All Rights Reserved                    //
+///////////////////////////////////////////////////////////////
+//                                                           //
+//       THIS IS UNPUBLISHED PROPRIETARY SOURCE CODE OF      //
+//        Agfa HealthCare N.V. and/or its affiliates.        //
+//      The copyright notice above does not evidence any     //
+//     actual or intended publication of such source code.   //
+//                                                           //
+///////////////////////////////////////////////////////////////
+//
+package org.dcm4che3.imageio.plugins.dcm;
+
+import org.dcm4che3.data.Attributes;
+import org.dcm4che3.data.Fragments;
+import org.dcm4che3.data.Tag;
+import org.dcm4che3.data.VR;
+import org.dcm4che3.image.ColorModelFactory;
+import org.dcm4che3.image.PhotometricInterpretation;
+import org.dcm4che3.imageio.codec.TransferSyntaxType;
+
+import javax.imageio.stream.ImageInputStream;
+import java.io.IOException;
+import java.net.URI;
+
+/**
+ * Interface that provides basic methods for accessing DICOM attribute data.
+ * @author Andrew Cowan (andrew.cowan@agfa.com)
+ */
+public interface DicomAccessor {
+
+    /**
+     * Get the base URI of the underlying DICOM data.
+     */
+    URI getURI();
+
+    /**
+     * Get the File Meta Information (FMI) header attributes.
+     * @return FMI attributes from the file.
+     */
+    Attributes getFileMetaInformation();
+
+    /**
+     * Get the header attributes for the instance.  This will not contain the PixelData tag,
+     * the {@link #openPixelStream(int)} method should be used to access pixel data.
+     * @return Header attributes for this instance.
+     */
+    Attributes getAttributes();
+
+    /**
+     * Open an ImageInputStream which is partitioned to view only the pixels for the indicated frameIndex.
+     * @param frameIndex 0 based index of the frame to view.
+     * @throws IOException if the pixel data source could not be accessed
+     * @throws IllegalArgumentException if there is no pixel data tag for this object.
+     * @return An open ImageInputStream.
+     */
+    // Do we want to have a PixelDataMissingException that subclasses IOException?
+    ImageInputStream openPixelStream(int frameIndex) throws IOException;
```

Comment:
This is the critical functionality that was extracted from DicomImageReader:  being able to find a frame or consolidate the fragments of an encapsulated single frame image.  ImageInputStream gives us an excellent class to do it with, and it's consistent with the rest of the module.